Tensile/Source/client/main.cpp

```diff
@@ -305,7 +305,7 @@ namespace Tensile
                 {
                     if(logLevel >= LogLevel::Verbose)
                         std::cout << "Loading " << filename << std::endl;
-                    adapter.loadCodeObjectFile(filename);
+                    HIP_CHECK_EXC(adapter.loadCodeObjectFile(filename));
```

Comment:
It looks like this is causing problems for some tests because of loading code objects for other architectures. We probably need to either ignore loading invalid code objects, or maybe filter them somehow.

Reply:
@AlexBrownAMD 
The precheck_in test of aldebaran failed due to that "xnack+" is enabled but original tensile didn't treat hipErrorSharedObjectInitFailed as a error.
I suggest that we should not ignore the loading error.
I file another commit to temporarily disable building xnack+ for aldebaran and test can pass.

@zaliu @yoichiyoshida 
The xnack+ commit is  https://github.com/ROCmSoftwarePlatform/Tensile/commit/92944f09209fe3677882a91164aa6b671b87c94a
Could you help to check why xnack+ cannot be loaded?
Before the root cause is identified, I suggest to disable it temporarily. https://github.com/ROCmSoftwarePlatform/Tensile/pull/1332/commits/e3718acb821fcd4cb62b23bbc25138c2341704f4 is in this PR.
You can enable AMD_LOG_LEVEL=3 and run pre_checkin test on aldebaran to see the errors.
test commands:

> AMD_LOG_LEVEL=3 tox -v -e py36 -- Tensile/Tests -m pre_checkin

Logs:
> Loading /workspace/Tensile/.tox/py36/client/1_BenchmarkProblems/Cij_Aik_Bjk_SB_00/00_Final/source/library/Kernels.so-000-gfx90a-xnack+.hsaco
> :3:hip_module.cpp           :58  : 519815182272 us: 54343: [7fb6627db180] hipModuleLoad ( 0x7fffa7719a30, /workspace/Tensile/.tox/py36/client/1_BenchmarkProblems/Cij_Aik_Bjk_SB_00/00_Final/source/library/Kernels.so-000-gfx90a-xnack+.hsaco )
> :3:devprogram.cpp           :2394: 519815182540 us: Using Code Object V4.
> :1:devprogram.cpp           :1698: 519815182647 us: Error: AMD HSA Code Object loading failed: HSA_STATUS_ERROR_INCOMPATIBLE_ARGUMENTS: The arguments passed to a functions are not compatible.
> 
> :3:hip_module.cpp           :60  : 519815182700 us: 54343: [7fb6627db180] hipModuleLoad: Returned hipErrorSharedObjectInitFailed :

Reply:
The system can be in either xnack+ or xnack- mode at any moment in time, never both.  As I mentioned in private, our test systems boot in xnack- mode; as such, the xnack+ code object loading at runtime will fail due to it being incompatible with the current xnack mode the system is in.

Reply:
Thanks your replies, I will remove the second commit :"disable xnack+" from this PR once the runtime can dynamically check/load correct xnack mode. 
I will set this PR as non-merge first.